src/components/tree/tree.component.js

```diff
@@ -96,6 +97,15 @@ function Tree() {
         showAnimation();
     }
 
+    const findStuff = () => {
+        nodeList = [];
+        if(treeNode.findingKey(treeNode.root, calculateTraversal, parseInt(findKey))){
+          showAnimation();
```

Comment:
Do we need to write this as a separate function? Can't we write it inside `traversal` function and pass optional `key` as an argument?

Reply:
actually in that way the time complexity would be O(n) and in the current way, the time complexity is O(log(n) ). Also, to avoid confusion, the traversal function will be needed to be duplicated. So i suggest to keep the current function. Please let me know your final call on the same.

Reply:
@sambhavjain9138  I was not talking about the logic. That is good. But can't we call this function in a switch case in the same traversal function?

Reply:
gotcha. will do so. Sorry for the confusion.

Regards,
Sambhav